### PR TITLE
refactor(`monitoring-provider`): count signatures for operator address

### PR DESCRIPTION
**Pull request by lubtd**
### NOTE: Don't mind the `reward` unit tests error, we need to implement the second part of the refactoring to fix these tests. This PR is not merged on `develop`

Part of https://github.com/tendermint/spn/issues/552

Add `staking` as dependency for monitoring provider and count signature for the operator address instead of directly the consensus address

To test: https://github.com/lubtd/orbit use the branch
Operator address can be visualized in the monitoring info (cons address is still here to keep some comptabilities)
```
MonitoringInfo:
  signatureCounts:
    blockCount: "5"
    counts:
    - RelativeSignatures: "1.333333333333333332"
      consAddress: null
      opAddress: cosmosvaloper17qqqsmyx43efr5ywp33h35l4dlmacfxvtkyed5
    - RelativeSignatures: "1.333333333333333332"
      consAddress: null
      opAddress: cosmosvaloper1kgjf5nyuk2288tqe540ujt8l2zrd83pygumegm
    - RelativeSignatures: "1.333333333333333332"
      consAddress: null
      opAddress: cosmosvaloper1v5a0gg2aeflfkzvjwqucu20arlawj0aywquu22
  transmitted: false
```

### CODE REVIEW

File: x/monitoringp/module.go [hunk 1]
```diff
@@ -169,10 +169,13 @@ func (AppModule) ConsensusVersion() uint64 { return 2 }
 // BeginBlock executes all ABCI BeginBlock logic respective to the capability module.
 func (am AppModule) BeginBlock(ctx sdk.Context, bb abci.RequestBeginBlock) {
 	// reports signatures for the block
-	am.keeper.ReportBlockSignatures(ctx, bb.LastCommitInfo, bb.Header.Height)
+	err := am.keeper.ReportBlockSignatures(ctx, bb.LastCommitInfo, bb.Header.Height)
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("error reporting block signatures %s", err.Error()))
```
Comment (aljo242) [inline x/monitoringp/module.go:174]:
```suggestion
		ctx.Logger().Error(fmt.Sprintf("error reporting block signatures: %s", err.Error()))
```

File: x/monitoringp/module.go [hunk 2]
```diff
@@ -169,10 +169,13 @@ func (AppModule) ConsensusVersion() uint64 { return 2 }
...
+	}
 
 	// check and transmit signatures
-	err := am.keeper.TransmitSignatures(ctx, bb.Header.Height)
+	err = am.keeper.TransmitSignatures(ctx, bb.Header.Height)
 	if err != nil {
 		ctx.Logger().Error(fmt.Sprintf("error transmitting the validator signatures %s", err.Error()))
```
Comment (aljo242) [inline x/monitoringp/module.go:180]:
```suggestion
		ctx.Logger().Error(fmt.Sprintf("error transmitting the validator signatures: %s", err.Error()))
```

File: x/monitoringp/keeper/begin_block_test.go [hunk 3]
```diff
@@ -3,64 +3,61 @@ package keeper_test
...
-		}
-
-		return lci
+	// valSet is a validator set for testing purpose that represent consensus addresses
+	// association with operator addresses
+	type val struct {
+		opAddr   string
+		consAddr []byte
 	}
+	type valSet []val
```
Comment (aljo242) [inline x/monitoringp/keeper/begin_block_test.go:22]: Type is unused.  Will this be used by a later PR?

**Review by aljo242 — CHANGES_REQUESTED** [review]

File: x/monitoringp/keeper/begin_block_test.go (hunk 3 above)
Comment (lubtd) [reply x/monitoringp/keeper/begin_block_test.go:22]: That some dust from a previous implementation thanks for spotting

**Review by aljo242 — APPROVED** [review]

**Pull request merged**